### PR TITLE
[MI-3239] Added testcases for connect API

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**

### CODE REVIEW

File: server/api_test.go [hunk 1]
```diff
@@ -742,3 +742,74 @@ func TestNeedsConnect(t *testing.T) {
...
+		Name               string
+		SetupPlugin        func(*plugintest.API)
+		SetupStore         func(*storemocks.Store)
+		ExpectedResult     string
+		ExpectedStatusCode int
+	}{
+		{
+			Name: "connect: User connected",
+			SetupPlugin: func(api *plugintest.API) {
+				api.On("KVSet", mock.AnythingOfType("string"), mock.AnythingOfType("[]uint8")).Return(nil)
```
Comment (ayusht2810) [inline server/api_test.go:757]: check if we can remove `mock.AnythingOfType("string")`

File: server/api_test.go [hunk 2]
```diff
@@ -742,3 +742,74 @@ func TestNeedsConnect(t *testing.T) {
...
+			r.Header.Add("Mattermost-User-ID", testutils.GetID())
+			plugin.ServeHTTP(nil, w, r)
+
+			result := w.Result()
+			defer result.Body.Close()
+
+			assert.NotNil(t, result)
+			assert.Equal(test.ExpectedStatusCode, result.StatusCode)
+
+			bodyBytes, _ := io.ReadAll(result.Body)
```
Comment (ayusht2810) [inline server/api_test.go:808]: check error here

File: server/api_test.go [hunk 3]
```diff
@@ -742,3 +742,74 @@ func TestNeedsConnect(t *testing.T) {
...
+			bodyString := string(bodyBytes)
```
Comment (ayusht2810) [inline server/api_test.go:809]: use this value directly

File: server/api_test.go [hunk 4]
```diff
@@ -742,3 +742,74 @@ func TestNeedsConnect(t *testing.T) {
...
+			ExpectedResult:     "Error trying to connect the account, please try again.\n",
+			ExpectedStatusCode: http.StatusInternalServerError,
+		},
+		{
+			Name: "connect: Error in storing the code verifier",
+			SetupPlugin: func(api *plugintest.API) {
+				api.On("KVSet", mock.AnythingOfType("string"), mock.AnythingOfType("[]uint8")).Return(&model.AppError{
+					Message: "error in storing the code verifier",
+				})
+				api.On("LogError", "Error in storing the code verifier", "error", "error in storing the code verifier").Return(nil)
```
Comment (ayusht2810) [inline server/api_test.go:781]:
```suggestion
				api.On("LogError", "Error in storing the code verifier", "error", "error in storing the code verifier").Return(nil).Times(1)
```

Comment (ayusht2810) [reply server/api_test.go:781]: add wherever applicable

File: server/api_test.go [hunk 5]
```diff
@@ -742,3 +743,75 @@ func TestNeedsConnect(t *testing.T) {
...
+			},
+			ExpectedResult:     "Error trying to connect the account, please try again.\n",
+			ExpectedStatusCode: http.StatusInternalServerError,
+		},
+	} {
+		t.Run(test.Name, func(t *testing.T) {
+			assert := assert.New(t)
+			plugin := newTestPlugin(t)
+
+			test.SetupPlugin(plugin.API.(*plugintest.API))
```
Comment (manojmalik20) [inline server/api_test.go:795]: Add `api.assertExpectations` .

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**